src/SFA.DAS.AdminService.Web/Validators/Roatp/Applications/RoatpFinancialClarificationViewModelValidator.cs

```diff
@@ -1,93 +1,158 @@
+                            when string.IsNullOrWhiteSpace(vm.InadequateComments):
+                            validationResponse.Errors.Add(new ValidationErrorDetail("InadequateComments",
+                                "Enter why the application was graded inadequate"));
+                            break;
+                        case FinancialApplicationSelectedGrade.Inadequate
+                            when HasExceededWordCount(vm.InadequateComments):
+                            validationResponse.Errors.Add(new ValidationErrorDetail("InadequateComments",
+                                "Your comments must be 500 words or less"));
+                            break;
                         case FinancialApplicationSelectedGrade.Outstanding:
-                    case FinancialApplicationSelectedGrade.Good:
-                    case FinancialApplicationSelectedGrade.Satisfactory:
-                        switch (vm.FinancialReviewDetails.SelectedGrade)
-                        {
-                            case FinancialApplicationSelectedGrade.Outstanding:
-                                ProcessDate(vm.OutstandingFinancialDueDate, "OutstandingFinancialDueDate", context);
-                                break;
-                            case FinancialApplicationSelectedGrade.Good:
-                                ProcessDate(vm.GoodFinancialDueDate, "GoodFinancialDueDate", context);
-                                break;
-                            case FinancialApplicationSelectedGrade.Satisfactory:
-                                ProcessDate(vm.SatisfactoryFinancialDueDate, "SatisfactoryFinancialDueDate", context);
-                                break;
-                        }
+                        case FinancialApplicationSelectedGrade.Good:
+                        case FinancialApplicationSelectedGrade.Satisfactory:
+                            switch (vm.FinancialReviewDetails.SelectedGrade)
+                            {
+                                case FinancialApplicationSelectedGrade.Outstanding:
+                                    ProcessDate(vm.OutstandingFinancialDueDate, "OutstandingFinancialDueDate",
+                                        validationResponse);
+                                    break;
+                                case FinancialApplicationSelectedGrade.Good:
+                                    ProcessDate(vm.GoodFinancialDueDate, "GoodFinancialDueDate", validationResponse);
+                                    break;
+                                case FinancialApplicationSelectedGrade.Satisfactory:
+                                    ProcessDate(vm.SatisfactoryFinancialDueDate, "SatisfactoryFinancialDueDate",
+                                        validationResponse);
+                                    break;
+                            }
 
-                        break;
-                }
-            });
+                            break;
+                    }
```

Comment:
Could this be done as separate method within the validator? The Validate method is quite lengthy otherwise